webapp-django/crashstats/crashstats/static/crashstats/js/socorro/correlation.js

```diff
@@ -97,4 +97,169 @@ $(document).ready(function () {
+    }
+
+    function toPercentage(num) {
+        let result = (num * 100).toFixed(2);
+
+        if (result == '100.00') {
+          return '100.0';
+        }
+
+        if (result.substring(0, result.indexOf('.')).length == 1) {
+          return '0' + result;
+        }
+
+        return result;
+    }
+
+  function confidenceInterval(count1, total1, count2, total2) {
+    let prop1 = count1 / total1;
+    let prop2 = count2 / total2;
+    let diff = prop1 - prop2;
+
+    // Wald 95% confidence interval for the difference between the proportions.
+    let standard_error = Math.sqrt(prop1 * (1 - prop1) / total1 + prop2 * (1 - prop2) / total2);
+    let ci = [diff - 1.96 * standard_error, diff + 1.96 * standard_error];
+
+    // Yates continuity correction for the confidence interval.
+    let correction = 0.5 * (1.0 / total1 + 1.0 / total2);
+
+    return [ci[0] - correction, ci[1] + correction];
+  }
+
+  function sortCorrelationData(correlationData, total_reference, total_group) {
+    return correlationData
+    .sort((a, b) => {
+      let rule_a_len = Object.keys(a.item).length;
+      let rule_b_len = Object.keys(b.item).length;
+
+      if (rule_a_len < rule_b_len) {
+        return -1;
+      }
+
+      if (rule_a_len > rule_b_len) {
+        return 1;
+      }
+
+      let ciA = confidenceInterval(a.count_group, total_group, a.count_reference, total_reference);
+      let ciB = confidenceInterval(b.count_group, total_group, b.count_reference, total_reference);
+
+      return Math.min(Math.abs(ciB[0]), Math.abs(ciB[1])) - Math.min(Math.abs(ciA[0]), Math.abs(ciA[1]));
+    });
+  }
+
+  function text(textElem, signature, channel, product) {
+    loadCorrelationData(signature, channel, product)
+    .then(data => {
+      textElem.text('');
+
+      if (!(product in data)) {
+        textElem.text('No correlation data was generated for the \'' + product + '\' product.');
+        return;
+      }
+
+      if (!(signature in data[product][channel]['signatures']) || !data[product][channel]['signatures'][signature]['results']) {
+        textElem.text('No correlation data was generated for the signature "' + signature + '" on the ' + channel + ' channel, for the \'' + product + '\' product.');
+        return;
+      }
+
+      let correlationData = data[product][channel]['signatures'][signature]['results'];
+
+      let total_reference = data[product][channel].total;
+      let total_group = data[product][channel]['signatures'][signature].total;
+
+      textElem.text(sortCorrelationData(correlationData, total_reference, total_group)
+      .reduce((prev, cur) =>
+        prev + '(' + toPercentage(cur.count_group / total_group) + '% in signature vs ' + toPercentage(cur.count_reference / total_reference) + '% overall) ' + itemToLabel(cur.item) + '\n'
+      , ''));
+    });
+  }
+
+  window.correlations = {};
+  window.correlations.loadCorrelationData = loadCorrelationData;
+  window.correlations.writeResults = text;
```

Comment:
Why these three lines? I don't get it. 
And perhaps it wasn't planned. 

So you're defining all of these functions AFTER the DOM has been loaded. Why can't the functions be define BEFORE or INDEPENDENT of the loading of the DOM?

Use the module pattern or just encapsulate all of these functions as methods in a class.